controllers/machine_controller.go

```diff
@@ -75,10 +78,18 @@ func (r *MachineReconciler) SetupWithManager(mgr ctrl.Manager, options controlle
 		WithOptions(options).
 		Build(r)
 
+	if err != nil {
+		return errors.Wrap(err, "failed setting up with a controller manager")
+	}
+
 	r.controller = c
 	r.recorder = mgr.GetEventRecorderFor("machine-controller")
 	r.config = mgr.GetConfig()
-	return err
+
+	r.scheme = runtime.NewScheme()
+	// Add the Kubernetes types into the scheme
+	err = scheme.AddToScheme(r.scheme)
+	return errors.Wrap(err, "failed adding Kubernetes types to the scheme")
```

Comment:
Maybe I'm a bit old school, but I tend to prefer an explicit check:

```go
if err != nil {
  return errors.Wrap(...)
}

return nil
```

Anyone have strong feelings on this?